test/unit/GoogleAnalyticsEventProcessor/getClientId_test.js

```diff
@@ -0,0 +1,142 @@
+goog.module('measurementLibrary.testing.eventProcessor.GoogleAnalyticsEventProcessor.getClientId');
+goog.setTestOnly();
+
+const DataLayerHelper = goog.require('dataLayerHelper.helper.DataLayerHelper');
+const uniqueId = goog.require('measurementLibrary.eventProcessor.generateUniqueId');
+const GoogleAnalyticsEventProcessor = goog.require('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * Mock storage interface for testing.
+ * @implements {StorageInterface}
+ */
+class MockStorage {
+  /**
+   * @param {!Object<string,*>=} mockStorage
+   */
+  constructor(mockStorage = {}) {
+    this.mockStorage = mockStorage;
+  }
+
+  /**
+   * @param {string} key
+   * @param {*} value
+   */
+  save(key, value) {
+    this.mockStorage[key] = value;
+  }
+
+  /**
+   * @param {string} key
+   * @return {*} value
+   */
+  load(key) {
+    return this.mockStorage[key];
+  }
+}
+
+/**
+ * Mock model interface for testing
+ */
+class MockModelInterface {
+  /**
+   * @param {!Object<string,*>} model
+   */
+  constructor(model) {
+    this.dataLayer = [model];
+    this.helper = new DataLayerHelper(this.dataLayer);
+  }
+
+  /**
+   * @param {string} key
+   * @return {*} value
+   */
+  get(key) {
+    return this.helper.get(key);
+  }
+
+  /**
+   * @param {string} key
+   * @param {*} value
+   */
+  set(key, value) {
+    this.dataLayer.push({[key]: value});
+  }
+}
+
+describe('The `getClientId_` method ' +
+    'of GoogleAnalyticsEventProcessor', () => {
+  const mockStoredId = 'stored_client_id';
+  const mockModelId = 'model_client_id';
+  const mockGeneratedId = 'generated_client_id';
+  const eventProcessor = new GoogleAnalyticsEventProcessor();
+  let mockStorage;
+  let mockModelInterface;
+  let generateUniqueId;
+
+  // mock the unique ID module
+  beforeAll(() => {
+    generateUniqueId = uniqueId.generateUniqueId;
+    uniqueId.generateUniqueId = () => {
+      return mockGeneratedId;
+    };
+  });
+
+  /**
+   * Builds expectation object for the `getClientId` function by passing
+   * in (if any) the client IDs stored in long term storage and the global
+   * data model.
+   * @param {string|undefined} storedClientId
+   * @param {string|undefined} modelClientId
+   * @return {!Expectation}
+   */
+  function expectClientId(
```

Comment:
I'm not sure I like this pattern. I think it could be better to reset mockStorage/mockModelInterface in a "beforeEach". Then just do a normal expect on the result of getClientId_.

This is somewhat controversial in the developer community at large, but generally Google prefers test logic to be explicit in most cases. I like the idea of having helper functions for complicated set up or complicated validation (separate), but this helper does setup and executes code under test (as well as calls the expect) in the same function.

For more info, see: 
https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html